cmd/snap-confine/udev-support.c

```diff
@@ -107,8 +107,12 @@ void run_snappy_app_dev_add(struct snappy_udev *udev_s, const char *path)
 
 	struct udev_device *d =
 	    udev_device_new_from_syspath(udev_s->udev, path);
-	if (d == NULL)
-		die("cannot find device from syspath %s", path);
+	// udev may be late with registering the device, e.g. under high load, and
+	// udev lookup will fail.
+	if (d == NULL) {
+		debug("cannot find device from syspath %s", path);
+		return;
+	}
 	dev_t devnum = udev_device_get_devnum(d);
```

Comment:
I was thinking, this can fail as well, on failure it returns major/minor of zero. Could you add a condition below, so that `_run_snappy_app_dev_add_majmin` doesn't get called in that case please?

Reply:
Indeed, it can fail although that would probably be a sign of a different problem. I'll add a check.

Reply:
Done